### DISCUSSION TITLE
Import from BoostNote legacy

### DISCUSSION THREAD

**Discussion by ecraig12345** [BoostIO/BoostNote.next-local#53] [Q&A] [answered]
Is there an easy way to import my notes from the legacy version of BoostNote? When I try to add the old storage using "Create or import Local Space" I get an error "TypeError: Cannot read property '/' of undefined."

**Comment by Komediruzecki** [comment, marked as answer]
Hi, 

Well, new boost note local space does not have this option currently (latest version), so the only way to easily import the notes would be to install 0.19.1 (older versions, you might need to use this repository releases for such versions: https://github.com/BoostIO/BoostNote-App/releases/tag/v0.22.0 - EDIT: the link goes to this repository release, not the one you need, go to this link for older version:  https://github.com/BoostIO/BoostNote-App/releases/tag/v0.19.0) and:

1. First create boost note next space (set location and name)
2. Then go to Preferences and chose the Space/Storage tab (you might need to refresh/create a note to see the Space preference tab)
3. There should be a form to import the notes into this new boost note next space from legacy storage

For more info, there should be some details and screenshots in this tutorial: https://medium.com/boostnote/how-to-migrate-data-from-old-boostnote-efd8a5c7aa7d

After you have successfully converted the legacy storage to boost note next storage, install the latest version of boost note local space from this repository and import it as boost note next storage.

**Reply by ecraig12345** [reply]
That worked. Thanks!

**Reply by jaques-sam** [reply]
Haha, I got this warning :-( 

```tag name is invalid, got `operator<<` ```
This means it's not able to convert C++ notes?

**Reply by Komediruzecki** [reply]
Can you explain the whole use case? You are converting from legacy? 
If there is such a restriction, you would probably need to remove invalid tags or copy those notes manually.

**Reply by jaques-sam** [reply]
Yes, I did an import of a legacy storage. Indeed, the tags were causing the issue! I had to remove special characters like <, > or #. Thx!
I still have something strange, when they are all converted to the cloud storage, the directories are gone. (I had 4 of them, only 1 remained).

**Reply by emcangi** [reply]
Hi, this answer isn't working for me. There is no option to migrate from legacy Boostnote in either 0.19, 0.20.1, 0.22, 0.14.1, etc. I did find it in 0.1 something, but that won't let me mass-move notes once they're imported, and doesn't recognize sub folders. (I'm using the Linux .deb)

What's the newest version where legacy Boostnote notes can still be imported for the Linux version?

**Reply by Komediruzecki** [reply]
It should work normally in 0.19.x, can you let me know which step you cannot reproduce?

You need to have space created and then import the legacy notes from settings, the space must be new boost note next space, file system type, no other.

**Reply by emcangi** [reply]
In v0.22, there is no option to designate file system type. When I log in to Boost Note next for the first time, all it does is ask me for a name and a domain: https://imgur.com/a/MKL1dFl

Then, when I go to the import section after entering a name but no domain, there is no legacy option.

**Reply by emcangi** [reply]
Sorry, misread--but v0.19 is the same!

**Reply by Komediruzecki** [reply]
That is correct, you cannot import legacy notes in v0.22!

You need to install the previous boost note next version, 0.19.x from here: https://github.com/BoostIO/BoostNote-App/releases/tag/v0.19.0

Once you do this, you can import legacy notes, by creating a boost note file system space and going to preferences.

After you have imported the notes, you can then go to local spaces (install latest from here: https://github.com/BoostIO/BoostNote.next-local/releases/tag/v0.22.3) and import the space there or migrate to the cloud from this repository latest release: https://github.com/BoostIO/BoostNote-App/releases/tag/v0.22.0

**Reply by emcangi** [reply]
okay, I've figured it out, but the app is a little confusing.

For anyone else reading: If you start Boost Note Next 0.19 for the first time, it asks you to make a space just like in the screenshot I linked, but there is *no way* to indicate that it should be local file storage. You have to go ahead and make it, go into Boost Note, then go File > Add Local Space, choose a local folder for it and make it, and then *delete the original space you made when you opened the app for the first time*. 

There should probably be more information on the splash screen at startup about this process!

Thanks!

**Reply by Komediruzecki** [reply]
Oh, sorry, I don't get that screen actually, mine is a bit different, in 0.19.0, I get a similar cloud space screen but there is also an option to create local space at the bottom, on your image - from the older intro screen, there is no such option, probably because app goes to cloud space by default.

Anyhow, you don't need to create cloud space to create a local space, you can do it from File | Add local space menu button.

This way you get the same screen and can create local space easily, but your method works as well.

We cannot really change the release backward, so this is the only way to do it. If you update the app regularly, you should get proper instructions on all of this, it is just a problem if you jump to many versions in the future.

Sorry for troubles.

**Reply by emcangi** [reply]
Maybe I had somehow managed to grab Boostnote.next instead of Boostnote.next-local, actually--that would explain why it didn't think there were any local options! Sorry also for the confusion!

**Comment by jaques-sam** [comment]
It is 0.19! If you make a local storage, in your preferences you’ll find the option to import a legacy storage.

**Reply by Komediruzecki** [reply]
Available again in latest release.

**Reply by jaques-sam** [reply]
I'm guessing this should be in the "Add space" menu option under "File", but with the latest release it just opens a white window:

![image](https://user-images.githubusercontent.com/42445148/167825432-5245fcab-06d1-460b-9fda-d809e1992d36.png)

**Reply by Komediruzecki** [reply]
> I'm guessing this should be in the "Add space" menu option under "File", but with the latest release it just opens a white window:
> 

Hi,

can you please describe your system? Is this on windows?

I just tried on Windows and previously on Linux, it seems to work just fine, can you try refreshing the window or closing/opening app and try again? And if this does not work, opening the developer tools console tab and see if there are any errors there? Was it a clean install or update, do you have any local spaces already loaded into the app?

It should open space creation form.
![image](https://user-images.githubusercontent.com/18196945/167953816-41a9aab4-60aa-42c7-8a8f-3a5c6eb34984.png)

**Comment by jaques-sam** [comment]
Yes, Windows 10. I did trying to refresh, open a new window etc. All the same result... I did an update of the app recently yes. And yes, I have already a local space (but the menu says "Add Space" :p

Update: When trying to open a document, creating a new space, or logout/login it takes ages to load. It does in the end (not sure if I'm connected) but it's unworkable. Opening a doc doesn't work (top bar takes forever to load until "request timed out"). So probably this has something to do with it :-) I added BoostNote with inbound+outbound rule in Windows firewall without success, maybe it's blocked by my company.

**Reply by Komediruzecki** [reply]
Hi,

it should not be blocked, but you are mentioning loading the page, documents, topbar, logout and login, this is not what boost note next local space application supports.

Are you maybe using boost note cloud? (https://github.com/BoostIO/BoostNote-App)
For boost note cloud, there was unexpected outage recently, now it should work, just try closing the app and opening it again, can you tell me which version you had installed?

For any other issue of using boost note cloud, please open an issue here: https://github.com/BoostIO/BoostNote-App/issues, or join our slack and ask there: https://join.slack.com/t/boostnote-group/shared_invite/zt-cun7pas3-WwkaezxHBB1lCbUHrwQLXw

**Reply by Komediruzecki** [reply]
P.S.

Yes, sorry, for boost note cloud application, Add space currently opens new blank window, not sure why, this is a bug, needs to be fixed.

**Comment by bigslipper** [comment]
Hi 
I have the same problem. When  i try to import an old archive for a tag with #  (ie "#rust_book") the import does'nt work.
I think that is a bug, I had solved by changing the cson's files presented in notes directory using a bash script.
